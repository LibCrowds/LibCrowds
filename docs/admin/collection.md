# Collection Admin

This section contains a series of pages via which users with admin rights can
manage the platform's collection microsites.

## Open Collection

Here you can select the collection microsite that you wish to manage. Click
'Open' to display the all additional settings pages for a microsite.

## New Collection

To create a new collection specify a name and description, then click 'Create'.

Following creation you will be given access to a range of additional
configuration options, which are described in more detail below.

## Settings

This is where the core settings for the microsite are configured, each of which
are explained below.

| Setting            | Description
|--------------------|------------------------------------------------------------------------------------------------|
| Published          | Once published, a microsite is added to the platform homepage and is accessible to all users.  |
| Name               | Applies branding throughout the collection microsite.                                          |
| Short name         | Used as an identifier and to form the microsite endpoint (e.g. `/collection/short_name`).      |
| Tagline            | Appears over the hero image on the microsite's homepage and forms a part of it's title.        |
| Description        | Appearson the microsite's homepage and forms its meta description.                             |
| Background Image   | The URL for an image used to provide a consistent theme for all of the microsite's core pages. |
| Logo               | The URL for a logo to be shown alongside the name on the microsite's homepage.                 |
| Forum URL          | The URL for the Discuss link in the microsite's main navigation bar.                           |
| Task Presenter     | The module used to present tasks for all of the microsite's projects.                          |
| Data Reuse License | The license by which all data collected via the microsite is made available.                   |

## Content

A set of markdown editors are provided here for modifying content for the About,
Projects and Data pages. Changes are made in real-time, although you may have
to refresh your browser to see them.

For the About page, navigation items will be generated from each level two
header, created in Markdown by using the ## tag, for example:

```
## Contact

This is the contact details section.
```

These navigation items can be clicked to scroll the users window to the top
of that section.

For more guidance on the GitHub flavoured markdown used, see this
[GitHub Guide](https://guides.github.com/features/mastering-markdown/).

## Terminology

You can use the options provided here to modify the domain object terminology
used throughout the collection microsite. For instance, task could be renamed
to objective, or task run to contribution.

## Tags

Tags are used to help users more easily location the types of project within a
collection that they're interested in. For each tag type created via this
admin section a new filter will be added to the microsite's Contribute
page (among others).

![Collection Admin - Tags](../assets/admin-collection-tags.png?raw=true)

To add a new tag click the 'Add tag' button at the top of the table. You
will then be given options to choose a tag type (e.g. Location), a tag name
(e.g. London) and a colour. Note that the colour applies to all tags of the
same type.

Once tags are created they can be added to individual projects via the
[Project Admin](admin/project.md) section.

Tags can be edited or removed by locating them in the table and clicking the
corresponding buttons. If you want to edit the color for a particular tag type
just click edit on any tags of that type.

## Featured

Here you can set the projects that appear on the Featured Projects section of
the collection microsite's homepage.

![Collection Admin - Featured](../assets/admin-collection-featured.jpg?raw=true)

## Delete

Here you can permanently delete a collection microsite.
