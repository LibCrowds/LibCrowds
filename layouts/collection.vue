<template>
  <div id="collection-base-layout">
    <div id="collection-background" :style="bgStyle"></div>
    <collection-navbar
      :collection-config="collectionConfig">
    </collection-navbar>
    <main>
      <nuxt
        :collection-config="collectionConfig">
      </nuxt>
    </main>
    <app-footer></app-footer>

    <notifications position="bottom left"></notifications>

  </div>
</template>

<script>
import localConfig from '@/local.config'
import CollectionNavbar from '@/components/navs/CollectionNavbar'
import AppFooter from '@/components/footers/App'

export default {
  head () {
    const tmpl = `%s - ${this.collectionConfig.name} | ${localConfig.brand}`
    return {
      titleTemplate: tmpl,
      meta: [
        {
          hid: 'description',
          name: 'description',
          content: this.collectionConfig.description
        },

        // Facebook Open Graph Markup
        { property: 'og:url', content: window.location.href },
        { property: 'og:title', content: this.collectionConfig.tagline },
        {
          property: 'og:description',
          content: this.collectionConfig.description
        }
      ]
    }
  },

  components: {
    CollectionNavbar,
    AppFooter
  },

  computed: {
    bgStyle () {
      const url = this.collectionConfig.bgImg
      return {
        background: `url(${url}) no-repeat center left /cover fixed`
      }
    },
    collectionConfig () {
      const collectionName = this.$route.params.collectionname
      return localConfig.collections[collectionName]
    }
  }
}
</script>

<style lang="scss" scoped>
#collection-base-layout {
  flex: 1 1 auto;
  display: flex;
  flex-direction: column;
  min-height: 100vh;

  #collection-background {
    z-index: -1;
    position: fixed;
    top: 0;
    left: 0;
    min-height: 100%;
    height: auto;
    width: 100%;
    min-width: 1024px;
    -webkit-background-size: cover;
    -moz-background-size: cover;
    -o-background-size: cover;
    background-size: cover;
  }

  main {
    flex: 1 1 auto
  }
}
</style>
