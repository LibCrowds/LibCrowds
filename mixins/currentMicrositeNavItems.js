/**
 * A mixin for the current collection microsite nav items.
 */
export const currentMicrositeNavItems = {
  computed: {
    currentMicrositeNavItems () {
      const micrositeShortName = this.$route.path.startsWith('/collection')
        ? this.$route.path.replace(/^(\/collection\/)/, '').split('/')[0]
        : null
      if (!micrositeShortName) {
        return []
      }
      return [
        {
          label: 'Home',
          link: {
            name: 'collection-short_name',
            params: {
              short_name: micrositeShortName
            }
          }
        },
        {
          label: 'About',
          link: {
            name: 'collection-short_name-about',
            params: {
              short_name: micrositeShortName
            }
          }
        },
        {
          label: 'Take Part',
          link: {
            name: 'collection-short_name-projects',
            params: {
              short_name: micrositeShortName
            }
          }
        },
        {
          label: 'Data',
          link: {
            name: 'collection-short_name-data',
            params: {
              short_name: micrositeShortName
            }
          }
        }
      ]
    }
  }
}
