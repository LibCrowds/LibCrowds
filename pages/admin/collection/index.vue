<template>
  <card-base :title="title" help="Manage the collection microsites">
    <b-btn
      slot="controls"
      variant="success"
      class="float-right"
      size="sm"
      :to="{
        name: 'admin-collection-new'
      }">
      New
    </b-btn>

    <infinite-loading-table
      ref="table"
      :fields="tableFields"
      no-border
      domain-object="category">
      <template slot="action" scope="category">
        <b-btn
          variant="success"
          size="sm"
          block
          exact
          :to="{
            name: 'admin-collection-short_name-settings',
            params: {
              short_name: category.item.short_name
            }
          }">
          Open
        </b-btn>
      </template>
    </infinite-loading-table>

  </card-base>
</template>

<script>
import PybossaForm from '@/components/forms/PybossaForm'
import CardBase from '@/components/cards/Base'
import InfiniteLoadingTable from '@/components/tables/InfiniteLoading'

export default {
  layout: 'admin-collection-dashboard',

  middleware: 'is-admin',

  data () {
    return {
      title: 'Open Collection',
      tableFields: {
        name: {
          label: 'Name'
        },
        created: {
          label: 'Created',
          class: 'text-center d-none d-md-table-cell',
          sortable: true
        }
      }
    }
  },

  head () {
    return {
      title: this.title
    }
  },

  components: {
    PybossaForm,
    CardBase,
    InfiniteLoadingTable
  },

  beforeMount () {
    this.$store.dispatch('UPDATE_CURRENT_COLLECTION', {})
  }
}
</script>
