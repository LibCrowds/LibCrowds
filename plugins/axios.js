export default function ({ $axios, error }) {
  $axios.onRequest(config => {
    config.headers['Content-Type'] = 'application/json'

    // Use any form data to set CSRF token header
    if (config.data && config.data.hasOwnProperty('csrf')) {
      config.headers['X-CSRFToken'] = config.data.csrf
    }

    // Ensure some data otherwise axois can delete Content-Type
    if (config.data === undefined) {
      config.data = {}
    }
    return config
  })

  $axios.onError(err => {
    err.code = parseInt(err.response && err.response.status) || 500
    err.message = (
      err.message ||
      err.response.statusText ||
      'Internal Server Error'
    )
    console.log(err)
    return error(err)
  })
}
