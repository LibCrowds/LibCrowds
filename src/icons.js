import Vue from 'vue'
import Icon from 'vue-awesome/components/Icon'
import users from 'ionicons/dist/svg/ios-people-outline.svg'
import projects from 'ionicons/dist/svg/ios-desktop-outline.svg'
import taskruns from 'ionicons/dist/svg/ios-create-outline.svg'

Vue.component('icon', Icon)

Icon.register({
  'users': {
    width: 512,
    height: 512,
    raw: users
  }
})

Icon.register({
  'taskruns': {
    width: 512,
    height: 512,
    raw: taskruns
  }
})

Icon.register({
  'projects': {
    width: 512,
    height: 512,
    raw: projects
  }
})
