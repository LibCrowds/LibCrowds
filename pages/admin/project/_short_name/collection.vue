<template>
  <card-base :title="title" :description="description">

    <b-form-input
      slot="controls"
      v-model="filter"
      class="search-control"
      size="sm"
      :placeholder="`Type to search by ${filterBy}`">
    </b-form-input>

    <infinite-loading-table
      ref="table"
      :fields="tableFields"
      :filter="filter"
      :filter-by="filterBy"
      domain-object="category">
      <template slot="action" scope="collection">
        <b-btn
          variant="success"
          size="sm"
          :disabled="collection.item.id == project.category_id"
          @click="setCollection(collection.item)">
          Select
        </b-btn>
      </template>
    </infinite-loading-table>
  </card-base>
</template>

<script>
import { notifications } from '@/mixins/notifications'
import { fetchProjectByName } from '@/mixins/fetchProjectByName'
import { metaTags } from '@/mixins/metaTags'
import InfiniteLoadingTable from '@/components/tables/InfiniteLoading'
import CardBase from '@/components/cards/Base'

export default {
  layout: 'admin-project-dashboard',

  mixins: [ fetchProjectByName, notifications, metaTags ],

  data () {
    return {
      title: 'Collection',
      description: 'Choose the collection that this project belongs to.',
      tableFields: {
        name: {
          label: 'Name',
          sortable: true
        },
        created: {
          label: 'Created',
          class: 'text-center d-none d-md-table-cell',
          sortable: true
        }
      },
      searchParams: {},
      updatedCollectionId: null,
      filter: null,
      filterBy: 'name'
    }
  },

  components: {
    InfiniteLoadingTable,
    CardBase
  },

  computed: {
    project () {
      return this.$store.state.currentProject
    }
  },

  methods: {
    /**
     * Set the collection for the project.
     * @param {Object} collection
     *   The collection.
     */
    setCollection (collection) {
      this.$axios.$put(`/api/project/${this.project.id}`, {
        category_id: collection.id
      }).then(data => {
        this.notifySuccess({ message: `Project moved to ${collection.name}` })
        this.$store.dispatch('UPDATE_CURRENT_PROJECT', data)
      }).catch(err => {
        this.$nuxt.error(err)
      })
    }
  }
}
</script>
