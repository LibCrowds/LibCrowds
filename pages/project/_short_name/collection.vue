<template>
  <b-card
    no-body>
    <div slot="header" class="mb-0">
      <h6 class="mb-0">{{ title }}</h6>
      <p class="text-muted mb-0">
        <small>
          Choose the collection microsite that this project should appear on.
        </small>
      </p>
    </div>

    <infinite-loading-table
      ref="table"
      :fields="tableFields"
      no-border
      domain-object="category">
      <template slot="action" scope="collection">
        <b-btn
          variant="success"
          size="sm"
          block
          :disabled="collection.item.id == project.category_id"
          @click="setCollection(collection.item)">
          Select
        </b-btn>
      </template>
    </infinite-loading-table>
  </b-card>
</template>

<script>
import { notifications } from '@/mixins/notifications'
import { fetchProjectByName } from '@/mixins/fetchProjectByName'
import InfiniteLoadingTable from '@/components/tables/InfiniteLoading'

export default {
  layout: 'project-dashboard',

  mixins: [ fetchProjectByName, notifications ],

  data () {
    return {
      title: 'Set Collection',
      tableFields: {
        name: {
          label: 'Name',
          sortable: true
        },
        created: {
          label: 'Created',
          class: 'text-center d-none d-md-table-cell',
          sortable: true
        }
      },
      searchParams: {},
      updatedCollectionId: null
    }
  },

  head () {
    return {
      title: this.title
    }
  },

  components: {
    InfiniteLoadingTable
  },

  computed: {
    project () {
      return this.$store.state.currentProject
    }
  },

  methods: {
    /**
     * Set the collection for the project.
     * @param {Object} collection
     *   The collection.
     */
    setCollection (collection) {
      this.$axios.$put(`/api/project/${this.project.id}`, {
        category_id: collection.id
      }).then(data => {
        this.notify({
          type: 'success',
          title: 'Success',
          message: `Project moved to ${collection.name}`
        })
        this.$store.dispatch('UPDATE_CURRENT_PROJECT', data)
      }).catch(err => {
        this.$nuxt.error({ statusCode: err.statusCode, message: err.message })
      })
    }
  }
}
</script>
