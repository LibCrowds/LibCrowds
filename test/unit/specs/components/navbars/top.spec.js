import Vuex from 'vuex'
import VueRouter from 'vue-router'
import BootstrapVue from 'bootstrap-vue'
import NuxtLink from '@/.nuxt/components/nuxt-link'
import Icon from 'vue-awesome/components/Icon'
import 'vue-awesome/icons/bars'
import 'vue-awesome/icons/caret-down'
import VueGravatar from 'vue-gravatar'
import { directive as onClickaway } from 'vue-clickaway'

import axios from 'axios'
import moxios from 'moxios'

import collection from '@/test/fixtures/collection.json'
import pbTestResponses from '@/test/fixtures/pybossaTestResponses.json'
import { mount, createLocalVue } from '@vue/test-utils'
import TopNavbar from '@/components/navbars/Top'
import { routes } from '@/test/fixtures/routes'
import { currentMicrositeNavItems } from '@/mixins/currentMicrositeNavItems'

describe('Top navbar', () => {
  let $axios = axios.create()
  let localVue = null
  let wrapper = null
  let store = null
  let router = null

  beforeEach(() => {
    localVue = createLocalVue()
    localVue.use(BootstrapVue)
    localVue.use(Vuex)
    localVue.use(VueRouter)
    localVue.component('icon', Icon)
    localVue.component('v-gravatar', VueGravatar)
    localVue.directive('on-clickaway', onClickaway)
    localVue.component(NuxtLink.name, NuxtLink)
    store = new Vuex.Store({
      state: {
        currentUser: {},
        currentCollection: {}
      }
    })
    router = new VueRouter({
      routes
    })

    moxios.install($axios)
  })

  afterEach(() => {
    moxios.uninstall($axios)
  })

  it('renders correctly with props', () => {
    wrapper = mount(TopNavbar, {
      localVue,
      store,
      router,
      mocks: {
        $axios
      },
      propsData: {
        fixedTop: true,
        navbarBrand: 'Foo',
        showHelp: true,
        transparent: true,
        dark: true
      }
    })
    const renderer = require('vue-server-renderer').createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('renders correctly when signed out and no collection', () => {
    wrapper = mount(TopNavbar, {
      localVue,
      store,
      router,
      mocks: {
        $axios
      }
    })
    const renderer = require('vue-server-renderer').createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('renders correctly when signed in and no collection', () => {
    const user = pbTestResponses.getAccount.data.user
    user.admin = false
    store.state.currentUser = user
    wrapper = mount(TopNavbar, {
      localVue,
      store,
      router,
      mocks: {
        $axios
      }
    })
    const renderer = require('vue-server-renderer').createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('renders correctly when signed out and has collection', () => {
    currentMicrositeNavItems.computed.currentMicrositeNavItems = () => {
      return [
        {
          label: 'Home',
          link: {
            name: 'collection-short_name',
            params: {
              short_name: collection.short_name
            }
          }
        },
        {
          label: 'About',
          link: {
            name: 'collection-short_name-about',
            params: {
              short_name: collection.short_name
            }
          }
        }
      ]
    }
    wrapper = mount(TopNavbar, {
      localVue,
      store,
      router,
      mocks: {
        $axios
      },
      mixins: [ currentMicrositeNavItems ]
    })
    const renderer = require('vue-server-renderer').createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })

  it('renders correctly when signed in and has collection', () => {
    const user = pbTestResponses.getAccount.data.user
    store.state.currentUser = user
    currentMicrositeNavItems.computed.currentMicrositeNavItems = () => {
      return [
        {
          label: 'Home',
          link: {
            name: 'collection-short_name',
            params: {
              short_name: collection.short_name
            }
          }
        },
        {
          label: 'About',
          link: {
            name: 'collection-short_name-about',
            params: {
              short_name: collection.short_name
            }
          }
        }
      ]
    }
    wrapper = mount(TopNavbar, {
      localVue,
      store,
      router,
      mocks: {
        $axios
      },
      mixins: [ currentMicrositeNavItems ]
    })
    const renderer = require('vue-server-renderer').createRenderer()
    renderer.renderToString(wrapper.vm, (err, str) => {
      expect(str).toMatchSnapshot()
    })
  })
})
