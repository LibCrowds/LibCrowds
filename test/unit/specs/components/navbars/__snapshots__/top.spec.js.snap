// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Top navbar renders correctly when signed in and has collection 1`] = `
<nav id="top-navbar" class="navbar navbar-light">
  <button type="button" class="btn px-2 border-top-0 border-left-0 btn-outline-dark">
    <svg version="1.1" role="presentation" width="13.714285714285714" height="16" viewBox="0 0 1536 1792" class="mr-1 fa-icon">
      <path d="M1536 1344v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45zM1536 832v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45zM1536 320v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45z"></path>
      <!---->
    </svg>
    <p class="mb-0">Menu</p>
  </button> <span class="top-navbar-left"><!----> <ul class="navbar-nav main-nav-menu d-none d-lg-flex"><li class="nav-item"><a href="#/collection/playbills" target="_self" class="nav-link">
        Home
      </a></li><li class="nav-item"><a href="#/collection/playbills/about" target="_self" class="nav-link">
        About
      </a></li></ul></span>
  <ul right="" class="navbar-nav top-navbar-right">
    <!---->
    <div id="announcements-list">
      <button id="announcements-toggle" type="button" class="btn nav-link btn-secondary">
        <svg version="1.1" role="presentation" width="16" height="16" viewBox="0 0 1792 1792" class="fa-icon">
          <path d="M912 1696q0-16-16-16-59 0-101.5-42.5t-42.5-101.5q0-16-16-16t-16 16q0 73 51.5 124.5t124.5 51.5q16 0 16-16zM1728 1408q0 52-38 90t-90 38h-448q0 106-75 181t-181 75-181-75-75-181h-448q-52 0-90-38t-38-90q50-42 91-88t85-119.5 74.5-158.5 50-206 19.5-260q0-152 117-282.5t307-158.5q-8-19-8-39 0-40 28-68t68-28 68 28 28 68q0 20-8 39 190 28 307 158.5t117 282.5q0 139 19.5 260t50 206 74.5 158.5 85 119.5 91 88z"></path>
          <!---->
        </svg>
        <!---->
      </button> <span style="display:none;"><div id="announcements-container" class="card dropdown-menu-right"><div class="card-header"><div>Notifications</div></div> <div id="announcements" class="custom-scrollbar"> <div class="m-1"><small>
            There are no notifications yet.
          </small></div></div></div></span></div>
    <li id="user-menu" class="pr-2 pl-1 nav-item b-nav-dropdown dropdown">
      <a href="#" id="user-menu__BV_button_" aria-haspopup="true" aria-expanded="false" class="nav-link">
        <div class="d-flex flex-row align-items-center">
          <div class="mr-1 small-avatar xs">
            <div class="base-avatar">
              <img src="//www.gravatar.com/avatar/6579e96f76baa00787a28653876c6127?s=100&amp;d=identicon&amp;r=g" alt="Avatar" alt-tag="Small thumbnail placeholder" class="img-fluid rounded-circle placeholder">
            </div>
          </div> <span class="d-none d-md-block mr-1">JohnDoe</span>
          <svg version="1.1" role="presentation" width="9.142857142857142" height="16" viewBox="0 0 1024 1792" class="fa-icon">
            <path d="M1024 704q0 26-19 45l-448 448q-19 19-45 19t-45-19l-448-448q-19-19-19-45t19-45 45-19h896q26 0 45 19t19 45z"></path>
            <!---->
          </svg>
        </div>
      </a>
      <div aria-labelledby="user-menu__BV_button_" class="dropdown-menu dropdown-menu-right">
        <a href="#/account/JohnDoe" role="menuitem" target="_self" class="dropdown-item">Profile
        </a>
        <a href="#/account/JohnDoe/settings/profile" role="menuitem" target="_self" class="dropdown-item">Settings
        </a>
        <div role="separator" class="dropdown-divider"></div>
        <a role="menuitem" href="#" target="_self" class="dropdown-item">Sign Out</a>
      </div>
    </li>
  </ul>
</nav>
`;

exports[`Top navbar renders correctly when signed in and no collection 1`] = `
<nav id="top-navbar" class="navbar navbar-light">
  <button type="button" class="btn px-2 border-top-0 border-left-0 btn-outline-dark">
    <svg version="1.1" role="presentation" width="13.714285714285714" height="16" viewBox="0 0 1536 1792" class="mr-1 fa-icon">
      <path d="M1536 1344v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45zM1536 832v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45zM1536 320v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45z"></path>
      <!---->
    </svg>
    <p class="mb-0">Menu</p>
  </button> <span class="top-navbar-left"><!----> <ul class="navbar-nav main-nav-menu d-none d-lg-flex"></ul></span>
  <ul right="" class="navbar-nav top-navbar-right">
    <!---->
    <div id="announcements-list">
      <button id="announcements-toggle" type="button" class="btn nav-link btn-secondary">
        <svg version="1.1" role="presentation" width="16" height="16" viewBox="0 0 1792 1792" class="fa-icon">
          <path d="M912 1696q0-16-16-16-59 0-101.5-42.5t-42.5-101.5q0-16-16-16t-16 16q0 73 51.5 124.5t124.5 51.5q16 0 16-16zM1728 1408q0 52-38 90t-90 38h-448q0 106-75 181t-181 75-181-75-75-181h-448q-52 0-90-38t-38-90q50-42 91-88t85-119.5 74.5-158.5 50-206 19.5-260q0-152 117-282.5t307-158.5q-8-19-8-39 0-40 28-68t68-28 68 28 28 68q0 20-8 39 190 28 307 158.5t117 282.5q0 139 19.5 260t50 206 74.5 158.5 85 119.5 91 88z"></path>
          <!---->
        </svg>
        <!---->
      </button> <span style="display:none;"><div id="announcements-container" class="card dropdown-menu-right"><div class="card-header"><div>Notifications</div></div> <div id="announcements" class="custom-scrollbar"> <div class="m-1"><small>
            There are no notifications yet.
          </small></div></div></div></span></div>
    <li id="user-menu" class="pr-2 pl-1 nav-item b-nav-dropdown dropdown">
      <a href="#" id="user-menu__BV_button_" aria-haspopup="true" aria-expanded="false" class="nav-link">
        <div class="d-flex flex-row align-items-center">
          <div class="mr-1 small-avatar xs">
            <div class="base-avatar">
              <img src="//www.gravatar.com/avatar/6579e96f76baa00787a28653876c6127?s=100&amp;d=identicon&amp;r=g" alt="Avatar" alt-tag="Small thumbnail placeholder" class="img-fluid rounded-circle placeholder">
            </div>
          </div> <span class="d-none d-md-block mr-1">JohnDoe</span>
          <svg version="1.1" role="presentation" width="9.142857142857142" height="16" viewBox="0 0 1024 1792" class="fa-icon">
            <path d="M1024 704q0 26-19 45l-448 448q-19 19-45 19t-45-19l-448-448q-19-19-19-45t19-45 45-19h896q26 0 45 19t19 45z"></path>
            <!---->
          </svg>
        </div>
      </a>
      <div aria-labelledby="user-menu__BV_button_" class="dropdown-menu dropdown-menu-right">
        <a href="#/account/JohnDoe" role="menuitem" target="_self" class="dropdown-item">Profile
        </a>
        <a href="#/account/JohnDoe/settings/profile" role="menuitem" target="_self" class="dropdown-item">Settings
        </a>
        <div role="separator" class="dropdown-divider"></div>
        <a role="menuitem" href="#" target="_self" class="dropdown-item">Sign Out</a>
      </div>
    </li>
  </ul>
</nav>
`;

exports[`Top navbar renders correctly when signed out and has collection 1`] = `
<nav id="top-navbar" class="navbar navbar-light">
  <button type="button" class="btn px-2 border-top-0 border-left-0 btn-outline-dark">
    <svg version="1.1" role="presentation" width="13.714285714285714" height="16" viewBox="0 0 1536 1792" class="mr-1 fa-icon">
      <path d="M1536 1344v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45zM1536 832v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45zM1536 320v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45z"></path>
      <!---->
    </svg>
    <p class="mb-0">Menu</p>
  </button> <span class="top-navbar-left"><!----> <ul class="navbar-nav main-nav-menu d-none d-lg-flex"><li class="nav-item"><a href="#/collection/playbills" target="_self" class="nav-link">
        Home
      </a></li><li class="nav-item"><a href="#/collection/playbills/about" target="_self" class="nav-link">
        About
      </a></li></ul></span>
  <ul right="" class="navbar-nav top-navbar-right">
    <a href="#/account/signin?next=%2F" target="_self" class="btn px-2 border-0 btn-outline-dark">
      <p class="mb-0">Sign in</p>
    </a>
    <a href="#/account/register" target="_self" class="btn px-2 border-0 btn-success">
      <p class="mb-0">Sign up</p>
    </a>
  </ul>
</nav>
`;

exports[`Top navbar renders correctly when signed out and no collection 1`] = `
<nav id="top-navbar" class="navbar navbar-light">
  <button type="button" class="btn px-2 border-top-0 border-left-0 btn-outline-dark">
    <svg version="1.1" role="presentation" width="13.714285714285714" height="16" viewBox="0 0 1536 1792" class="mr-1 fa-icon">
      <path d="M1536 1344v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45zM1536 832v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45zM1536 320v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45z"></path>
      <!---->
    </svg>
    <p class="mb-0">Menu</p>
  </button> <span class="top-navbar-left"><!----> <ul class="navbar-nav main-nav-menu d-none d-lg-flex"></ul></span>
  <ul right="" class="navbar-nav top-navbar-right">
    <a href="#/account/signin?next=%2F" target="_self" class="btn px-2 border-0 btn-outline-dark">
      <p class="mb-0">Sign in</p>
    </a>
    <a href="#/account/register" target="_self" class="btn px-2 border-0 btn-success">
      <p class="mb-0">Sign up</p>
    </a>
  </ul>
</nav>
`;

exports[`Top navbar renders correctly with props 1`] = `
<nav id="top-navbar" class="navbar navbar-dark bg-transparent fixed-top">
  <button type="button" class="btn px-2 border-top-0 border-left-0 btn-outline-dark">
    <svg version="1.1" role="presentation" width="13.714285714285714" height="16" viewBox="0 0 1536 1792" class="mr-1 fa-icon">
      <path d="M1536 1344v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45zM1536 832v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45zM1536 320v128q0 26-19 45t-45 19h-1408q-26 0-45-19t-19-45v-128q0-26 19-45t45-19h1408q26 0 45 19t19 45z"></path>
      <!---->
    </svg>
    <p class="mb-0">Menu</p>
  </button> <span class="top-navbar-left"><a href="#" target="_self" class="navbar-brand d-none d-lg-block ml-1 mr-2"><span>Foo</span></a>
  <ul class="navbar-nav main-nav-menu d-none d-lg-flex"></ul>
  </span>
  <ul right="" class="navbar-nav top-navbar-right">
    <a href="#/account/signin?next=%2F" target="_self" class="btn px-2 border-0 btn-outline-dark">
      <p class="mb-0">Sign in</p>
    </a>
    <a href="#/account/register" target="_self" class="btn px-2 border-0 btn-outline-dark">
      <p class="mb-0">Sign up</p>
    </a>
  </ul>
</nav>
`;
